backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,677 @@
+
+		require.NoError(writer.Close())
+
+		var errResponse string
+		err = requests.
+			URL(s.httpAddress() + "/v1alpha1/transforms").
+			Put().
+			BodyBytes(body.Bytes()).
+			ContentType(writer.FormDataContentType()).
+			AddValidator(requests.ValidatorHandler(
+				requests.CheckStatus(http.StatusCreated), // Allows 2xx otherwise
+				requests.ToString(&errResponse),
+			)).
+			Fetch(ctx)
+		require.Error(err)
+		assert.NotEmpty(errResponse)
+		assert.Truef(requests.HasStatusErr(err, http.StatusBadRequest), "Expected response status to be 400 (BAD REQUEST)")
+		assert.Contains(errResponse, "REASON_INVALID_INPUT")
+		assert.Contains(errResponse, "output_topic_names")
+	})
+
+	t.Run("try to deploy transform with an invalid request - protected env var (http)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 12*time.Second)
+		defer cancel()
+
+		body := &bytes.Buffer{}
+		writer := multipart.NewWriter(body)
+
+		inputTopicName := "wasm-tfm-create-test-i"
+		deployRequestMetadata := deployTransformRequest{
+			Name:             "deploy-transform-incomplete-metadata",
+			InputTopicName:   inputTopicName,
+			OutputTopicNames: []string{"some-random-topic"},
+			EnvironmentVariables: []KeyVal{{
+				Key:   "REDPANDA_KEYS_ARE_PROTECTED", // This should be flagged
+				Value: "bar",
+			}},
+		}
+
+		metadataPart, err := writer.CreateFormField("metadata")
+		require.NoError(err)
+		require.NoError(json.NewEncoder(metadataPart).Encode(deployRequestMetadata))
+
+		wasmPart, err := writer.CreateFormFile("wasm_binary", "transform.wasm")
+		require.NoError(err)
+
+		_, err = wasmPart.Write(identityTransform)
+		require.NoError(err)
+
+		require.NoError(writer.Close())
+
+		var errResponse string
+		err = requests.
+			URL(s.httpAddress() + "/v1alpha1/transforms").
+			Put().
+			BodyBytes(body.Bytes()).
+			ContentType(writer.FormDataContentType()).
+			AddValidator(requests.ValidatorHandler(
+				requests.CheckStatus(http.StatusCreated), // Allows 2xx otherwise
+				requests.ToString(&errResponse),
+			)).
+			Fetch(ctx)
+		require.Error(err)
+		assert.NotEmpty(errResponse)
+		assert.Truef(requests.HasStatusErr(err, http.StatusBadRequest), "Expected response status to be 400 (BAD REQUEST)")
+		assert.Contains(errResponse, "REASON_INVALID_INPUT")
+		assert.Contains(errResponse, "reserved")
+	})
+
+	t.Run("try to deploy transform with an invalid request - no wasm binary (http)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 12*time.Second)
+		defer cancel()
+
+		body := &bytes.Buffer{}
+		writer := multipart.NewWriter(body)
+
+		inputTopicName := "wasm-tfm-create-test-i"
+		deployRequestMetadata := deployTransformRequest{
+			Name:             "deploy-transform-incomplete-metadata",
+			InputTopicName:   inputTopicName,
+			OutputTopicNames: []string{"some-random-topic"},
+			EnvironmentVariables: []KeyVal{{
+				Key:   "READPANDA_KEYS_ARE_PROTECTED", // This should be flagged
```

Comment:
since this is a test for "no wasm binary" in requests, this env var, and comment, is a little confusing. The env var is "OK" because it's R**E**ADPANDA_KEYS_ARE_PROTECTED instead of `RADPANDA_KEYS_ARE_PROTECTED`, (no **E**), but the comment is still "// This should be flagged". I think generally this part can be cleaned up maybe, just so the env key is "FOO", and no comment?

Reply:
Haha two issues in one place, indeed it should not be there at all.